quota/redis/redisqm/manager.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package redisqm defines a Redis-based quota.Manager implementation.
+package redisqm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/trillian/quota"
+	"github.com/google/trillian/quota/redis/redistb"
+)
+
+// ParameterFunc is a function that should return a token bucket's parameters
+// for a given quota specification.
+type ParameterFunc func(spec quota.Spec) (capacity int, rate float64)
+
+// ManagerOptions holds the parameters for a Manager.
+type ManagerOptions struct {
+	// Parameters should return the parameters for a given quota.Spec. This
+	// value must not be nil.
+	Parameters ParameterFunc
+
+	// Prefix is a static prefix to apply to all Redis keys; this is useful
+	// if running on a multi-tenant Redis cluster.
+	Prefix string
+}
+
+type Manager struct {
```

Comment:
Could you add a type comment on here (since it's exported)?

Personally, I quite like the `var _ quota.Manager = &Manager{}` style "compiler documentation" that makes it locally clear what this type will be used for.

Reply:
Done 👍